Kinder/Registration.xaml.cs

```diff
@@ -27,9 +27,9 @@ public RegistrationWindow()
             InitializeComponent();
         }
 
-        public delegate void CheckDelegate(string Text);
+        public delegate void CheckDelegate<T> (T item);
         
-        static void Checks (string text, CheckDelegate input, CheckDelegate field, CheckDelegate exists)
+        static void Checks (string text, CheckDelegate<string> input, CheckDelegate<string> field, CheckDelegate<string> exists)
```

Comment:
it's not useful to add just the strings, but I guess there isn't a way to add different types

Reply:
That's how you do generics and we have no other delegates to make generic.